definitions/npm/lodash_v4.x.x/flow_v0.104.x-/test_lodash-v4.x.x.js

```diff
@@ -59,11 +76,46 @@ import xorBy from "lodash/xorBy";
 import zip from "lodash/zip";
 import zipWith from "lodash/zipWith";
 
+// Ideally, there should be a test for each of the variables below. To verify that these use cases do not break on code changes.
+// test arrays
 type ReadOnlyArray = $ReadOnlyArray<number>
-const readOnlyArray : ReadOnlyArray = [1, 2, 3, 4];
+const readOnlyArray : ReadOnlyArray = Object.freeze([1, 2, 3, 4]);
 
-type ReadOnlyObject = $ReadOnly<{ [string]: number, ... }>
-const readOnlyObject : ReadOnlyObject = { a: 1, b: 2, c: 3 };
+type ArrayOrNull = number[] | null;
+const arrayOrNull: ArrayOrNull = readOnlyArray[0] > 100 ? [1, 2, 3] : null;
+
+type ArrayOrVoid = number[] | void;
+const arrayOrVoid: ArrayOrVoid = readOnlyArray[0] > 100 ? [1, 2, 3] : undefined;
+
+// test objects
+type IndexerObject = { [string]: number, ... };
+const indexerObject: IndexerObject = { a: 1, b: 2, c: 3 };
+
+type ExactObject = {| a: number, b: number, c: number; |};
+const exactObject: ExactObject = { a: 1, b: 2, c: 3 };
+
+type ExactHeterogeneousObject = {| a: number, b: string, c: boolean; |};
+const exactHeterogeneousObject: ExactHeterogeneousObject = { a: 1, b: 'abc', c: true };
+
+type ReadOnlyIndexerObject = $ReadOnly<IndexerObject>
+const readOnlyIndexerObject: ReadOnlyIndexerObject = { ...indexerObject }; // example of inexact object
+
+type ReadOnlyExactObject = $ReadOnly<ExactObject>
+const readOnlyExactObject: ReadOnlyExactObject = { ...exactObject };
+
+type ObjectOrNull = IndexerObject | null;
+const objectOrNull: ObjectOrNull = readOnlyArray[0] > 100 ? indexerObject : null;
+
+type ObjectOrVoid = IndexerObject | void;
+const objectOrVoid: ObjectOrVoid = readOnlyArray[0] > 100 ? indexerObject : undefined;
+
+opaque type OpaqueKey = number;
+const toOpaqueKey = (v: number): OpaqueKey => v;
+type ObjectWithOpaqueKey = { [OpaqueKey]: string, ... };
+const objectWithOpaqueKey: ObjectWithOpaqueKey  = { [toOpaqueKey(1)]: 'a', [toOpaqueKey(2)]: 'b' };
```

Comment:
Unfortunately, I cannot test errors for opaque types properly, because these types are opaque only "outside of the file in which they are defined". 
I've tried to create a sibling file, but it says "module is not found" for 
`import {type OpaqueKey } from './test-opaque-types-v4.x.x'


So, testing for example that `forEach(opaqueObject, (v, k: string) => !k);`